app/assets/javascripts/discourse/app/components/post.gjs

```diff
@@ -0,0 +1,559 @@
+      return;
+    }
+
+    await post.get("topic.postStream").filterReplies(post.post_number, post.id);
+    this.topicController.updateQueryParams();
+  }
+
+  @action
+  async toggleLike() {
+    const post = this.args.post;
+    const likeAction = post.likeAction;
+
+    if (likeAction?.canToggle) {
+      const result = await likeAction.togglePromise(post);
+
+      this.appEvents.trigger("page:like-toggled", post, likeAction);
+      return this.#warnIfClose(result);
+    }
+  }
+
+  @action
+  async toggleReplyAbove(goToPost = false) {
+    const replyPostNumber = this.args.post.reply_to_post_number;
+
+    if (this.siteSettings.enable_filtered_replies_view) {
+      await this.args.post.topic?.postStream?.filterUpwards?.(
+        this.args.post.id
+      );
+      this.topicController.updateQueryParams();
+    }
+
+    const topicUrl = this.args.post.topicUrl;
+
+    // jump directly on mobile
+    if (this.site.mobileView) {
+      if (topicUrl) {
+        DiscourseURL.routeTo(`${topicUrl}/${replyPostNumber}`);
+      }
+
+      return;
+    }
+
+    if (this.repliesAbove?.value.length) {
+      this.repliesAbove = null;
+
+      if (goToPost === true) {
+        const { post_number } = this.args.post;
+        DiscourseURL.routeTo(`${topicUrl}/${post_number}`);
+      }
+    } else {
+      this.repliesAbove = new TrackedAsyncData(this.#loadRepliesAbove());
+    }
+  }
+
+  @action
+  async toggleReplies() {
+    return this.filteredRepliesView
+      ? await this.toggleFilteredRepliesView()
+      : await this.toggleRepliesBelow();
+  }
+
+  @action
+  toggleRepliesBelow(goToPost = false) {
+    if (this.repliesBelow.length) {
+      // since repliesBelow is a tracked array, let's truncate it instead of creating another one
+      this.repliesBelow.length = 0;
+
+      if (goToPost === true) {
+        const { topicUrl, post_number } = this.args.post;
+        DiscourseURL.routeTo(`${topicUrl}/${post_number}`);
+      }
+    } else {
+      return this.loadMoreReplies();
+    }
+  }
+
+  #warnIfClose(result) {
+    if (!result || !result.acted) {
+      return;
+    }
+
+    const lastWarnedLikes = this.keyValueStore.get("lastWarnedLikes");
+
+    // only warn once per day
+    const yesterday = Date.now() - 1000 * 60 * 60 * 24;
+    if (lastWarnedLikes && parseInt(lastWarnedLikes, 10) > yesterday) {
+      return;
+    }
+
+    const { remaining, max } = result;
+    const threshold = Math.ceil(max * 0.1);
+
+    if (remaining === threshold) {
+      this.dialog.alert(i18n("post.few_likes_left"));
+      this.keyValueStore.set({ key: "lastWarnedLikes", value: Date.now() });
+    }
+  }
+
+  async #loadRepliesAbove() {
+    const replies = await this.store.find("post-reply-history", {
+      postId: this.args.post.id,
+    });
+
+    return replies.map((reply) => this.store.createRecord("post", reply));
+  }
+
+  <template>
+    {{#unless @cloaked}}
+      <article
+        {{parentClass
```

Comment:
Using this modifier is a temporary workaround to avoid inserting a wrapping `div` when using `RenderGlimmer` in the post stream widget.

Once the post stream widget is glimmerified we can drop this and move the outer div into this component to add the classes directly

Reply:
RenderGlimmer provides an API for this already. Instead of a custom modifier, you can use `@args.setWrapperElementAttrs` as a helper:

```hbs
{{@args.setWrapperElementAttrs (hash class=(concatClass ...))}}
```

([implementation](https://github.com/discourse/discourse/blob/7f9a0bb2a75217731de3a0168302b66453152c09/app/assets/javascripts/discourse/app/widgets/render-glimmer.js#L165C25-L165C25))

Reply:
We're still using the modifier because of the tests but it was moved to the component file to avoid other components using this pattern